submissions/dashakim/oop-task/js/app.js

```diff
@@ -0,0 +1,100 @@
+//game score
+var score = 0
+//Enemy class with update(), render() methods
+var Enemy = function (x, y, speed) {
+    this.x = x
+    this.y = y
+    this.speed = speed
+    this.sprite = 'images/car.png';
+};
+var END_CANVAS = 510
+var BEGIN_CANVAS = -50
+Enemy.prototype.update = function (dt) {
+
+    this.x += this.speed * dt
+    if (this.x > END_CANVAS) {
+        this.x = BEGIN_CANVAS
+        //random speed for enemies
+        this.speed = 100 + Math.floor(Math.random() * 222)
+    }
+    // return to start position player when collide with enemy
+    if (player.x > (this.x - 60) && player.x < (this.x + 60) && player.y > (this.y - 60) && player.y < (this.y + 60)) {
+        player.startPosition();
```

Comment:
So your class `Enemy` internal behaviour relies on a global variable. If we decide to split the code base into modules (e.g. `enemy.js`, `player.js`, `app.js`) then this becomes a mystery.
Let every enemy know the player at enemy instance initialization and keep the reference to player inside enemy instance.

Reply:
I'm agree with your point of view, but I'm not sure that I fixed it correctly